### ISSUE TITLE
OOB accesses into Devito generated code

### ISSUE THREAD

**Issue by speglich** [devitocodes/devito#1215]
This issue occurs in the commit: 79f012799aa32c21e327c848dbeb61e20ab001a8

The code attached makes the Devito compiler generate a code with OOB accesses. 

The loop bounds are not computed correctly by Devito. Basically the div(1/rho grad(p) makes the stencil twice as wide as the space_order and the halo isn’t big enough so end up in an OOB that isn’t detected.

This issue is divided in two mainly steps:
 
* Find why OOB wasn't detected and fix it.

[example.txt](https://github.com/devitocodes/devito/files/4441495/example.txt)

**Comment by FabioLuporini** [comment]
I agree about the first step

I'm not sure about the latter step. I don't think we need to fix boundaries in the generated code, but rather implement stencil auto-wrapping. Let's discuss about this once the first issue is dealt with !

Thanks a lot for this

**Comment by speglich** [comment]
Just for documentation: 

@FabioLuporini said:

I investigated the segfault issue (I didn't mean to overstep you @speglich, but given @OscarM's comments here: https://github.com/devitocodes/devito/pull/1212#issuecomment-611201569 we should fix this ASAP because this appears to be much deeper than expected). Here are my findings:

Mathias, Joao: the reason Devito doesn't automatically shrink the x_m/x_M bounds is that detect_oob is called at the beginning of compilation, well before the OOB is generated
the OOB is caused by what formerly we were calling dse=aggressive mode, that is now default when you run an Operator. Plus, the changes I made to one compilation passes (Mathias: CIRE), are causing the OOB. Still investigating exactly why that is happening.

Mathias: we said we were expecting the "+ 20" offsets in the stencils due to the given space order and taken derivatives, why that is not happening still escapes my mind. The OOB (the "+ 20") is a by-product of a compilation pass, which as I said I'm still investigating.
